File: files-app/src/main/java/l/files/fs/local/Group.java
package l.files.fs.local;

import auto.parcel.AutoParcel;

/**
 * @see <a href="http://pubs.opengroup.org/onlinepubs/7908799/xsh/grp.h.html">grp.h</a>
 */
@AutoParcel
abstract class Group extends Native {

    Group() {
    }

    public abstract String getName();

    public abstract int getGid();

    public static Group create(String name, int gid) {
        return new AutoParcel_Group(name, gid);
    }

    static {
        init();
    }

    private static native void init();

    /**
     * @see <a href="http://pubs.opengroup.org/onlinepubs/7908799/xsh/getgrgid.html">getgrid()</a>
     */
    public static native Group getgrgid(int gid) throws ErrnoException;

}

File: files-app/src/main/kotlin/fs/local/Group.kt
package l.files.fs.local

import kotlin.platform.platformStatic

/**
 * http://pubs.opengroup.org/onlinepubs/7908799/xsh/grp.h.html
 */
private data class Group(val name: String, val gid: Int) : Native() {

    public companion object {

        init {
            init()
        }

        platformStatic native private fun init()

        /**
         * http://pubs.opengroup.org/onlinepubs/7908799/xsh/getgrgid.html
         */
        throws(javaClass<ErrnoException>())
        platformStatic native fun getgrgid(gid: Int): Group

    }

}
